src/iterative_sorting/iterative_sorting.py

```diff
@@ -3,28 +3,62 @@ def selection_sort( arr ):
     # loop through n-1 elements
     for i in range(0, len(arr) - 1):
         cur_index = i
-        smallest_index = cur_index
+        smallest_index = cur_index 
         # TO-DO: find next smallest element
-        # (hint, can do in 3 loc) 
-             
-
-
-
-        # TO-DO: swap
-
+        # (hint, can do in 3 loc)
+        for j in range(i, len(arr)):
+            
+            if (arr[j] < arr[smallest_index]):
+                
+                smallest_index = j
+            
 
+            # TO-DO: swap
+        c = arr[cur_index]
+        arr[cur_index] = arr[smallest_index]
+        arr[smallest_index] = c
+        #OR
+        #arr[smallest_index], arr[cur_index] = arr[cur_index], arr[smallest_index]
 
 
     return arr
 
 
 # TO-DO:  implement the Bubble Sort function below
 def bubble_sort( arr ):
+    for i in range(len(arr)):
```

Comment:
Think about how you can optimize this algo so that it doesn't keep iterating through an array once it's sorted. How can you check that it's sorted on each loop?